tests/Orckestra.Composer.Cart.Tests/Factory/Order/OrderDetailsViewModelFactory_CreateViewModel.cs

```diff
@@ -223,5 +196,84 @@ protected Overture.ServiceModel.Orders.Order CreateOrder(string orderStatus, str
 
         }
 
+        [Test]
+        [TestCase("New|Pending", new string[] { "New" }, true)]
+        [TestCase("New|Pending", new string[] { "Pending" }, true)]
+        [TestCase("New|Pending", new string[] { "New", "Pending" }, true)]
+        [TestCase("New|Pending", new string[] { "Canceled" }, false)]
+        [TestCase("New|Pending", new string[] { "New,Canceled" }, false)]
+        [TestCase(null, new string[] { "New", "Canceled" }, false)]
+        [TestCase("New|Pending", null, false)]
+        public void WHEN_valid_request_EditableShipmentStates_SHOULD_return_TRUE(string editableShipmentStates,
+            string[] cartShipmentStatuses,
+            bool expectedIsOrderEditable)
+        {
+            //Arrange
+            var factory = _container.CreateInstance<OrderDetailsViewModelFactory>();
+
+            //Act
+            var shipmentList = new List<Shipment>();
+            cartShipmentStatuses?.ToList().ForEach(
+                item => shipmentList.Add(new Shipment()
+                {
+                    Status = item ,
+                    Address = new Address(),
+                    FulfillmentMethod = new FulfillmentMethod
+                    {
+                        Cost = GetRandom.Double()
+                    },
+                    Taxes = new List<Tax>()
+                }));
```

Comment:
to create the List you can use Select instead of Foreach/Add
for example you can rewrite this like this:
`var shipmentList = cartShipmentStatuses?.Select( status => new Shipment() {} .... `